bundles/go/go_fmt.moon

```diff
@@ -0,0 +1,58 @@
+import app, config, mode, signal from howl
+import Process from howl.io
+
+append = table.insert
+
+run_command = (file) ->
+  args = {}
+  for arg in config.go_fmt_command\gmatch '%S+'
+    append args, arg
+  append args, file
+  success, out, err, p = pcall Process.execute, args
+  unless success
+    log.error out
+    return
+  unless p.successful
+    log.error err
+    return
+
+reload_buffer = (buffer) ->
+  buffer.text = buffer.file.contents
+  buffer.sync_etag = buffer.file.etag
+  buffer.modified = false
+  
+calculate_new_pos = (pos, before, after) ->
+  new_pos = 1
+  i = 1
+  while i <= pos and new_pos <= #after
+    if before[i] == after[new_pos]
+      i += 1
+      new_pos += 1
+    elseif before[i]\match '%s'
+      i += 1
+    else
+      new_pos += 1
+  new_pos-1
+
+fmt = (buffer) ->
+  return unless buffer.mode.name == 'go'
+
+  file = buffer.file
+  before = file.contents
+  run_command file
```

Comment:
Ack, so having troubles doing it this way. Because the signal emit happens asynchronously it completes before calling the external process. Which means it just modifies the buffer text after the save has happened.

Seems like the 'pre-save' hook would have to be able to block the save.

Not sure how best to proceed, ideas?